Source/relay/TourGuide/Items of the Month/2021/Daylight Shavings Helmet.ash

```diff
@@ -1,93 +1,90 @@
-		tooltip_table.listAppend(listMake("Barbell", "+25 Muscle and +50% Gear drops"));
-		tooltip_table.listAppend(listMake("Grizzly", "+25-50 MP regen and +5 " + HTMLGenerateSpanFont("Cold", "blue") + " resist"));
-		tooltip_table.listAppend(listMake("Surrealist", "+25 Mysticality and +50% Food drops"));
-		tooltip_table.listAppend(listMake("Musician", "+25 Moxie and +50% Booze drops"));
-		tooltip_table.listAppend(listMake("Gull-Wing", "+100% init and +5 " + HTMLGenerateSpanFont("Hot", "red") + " resist"));
-		tooltip_table.listAppend(listMake("Warlord", "+25 Weapon damage and +10% Crit"));
-		tooltip_table.listAppend(listMake("Wizard", "+25 Spell damage and +10% Spell Crit"));
-		tooltip_table.listAppend(listMake("Cowboy", "+25 Ranged damage and +50 maximum HP"));
-		tooltip_table.listAppend(listMake("Friendly", "+100% meat and +5 " + HTMLGenerateSpanFont("Sleaze", "purple") + " resist"));
-		tooltip_table.listAppend(listMake("Spectacle", "+50% item and +5 " + HTMLGenerateSpanFont("Spooky", "grey") + " resist"));
-			
-		// Map buff number to buff result
-		switch (nextBeardBuff)			
-		{
-			case 1:
-				nextBeardBuffEffect = "Toiletbrush, +25 ML and +5 " + HTMLGenerateSpanFont("Stench", "green") + " resist"; break;
-			case 2:
-				nextBeardBuffEffect = "Barbell, +25 Muscle and +50% Gear drops"; break;
-			case 3:
-				nextBeardBuffEffect = "Grizzly, +25-50 MP regen and +5 " + HTMLGenerateSpanFont("Cold", "blue") + " resist"; break;
-			case 4:
-				nextBeardBuffEffect = "Surrealist, +25 Mysticality and +50% Food drops"; break;
-			case 5:
-				nextBeardBuffEffect = "Musician, +25 Moxie and +50% Booze drops"; break;
-			case 6:
-				nextBeardBuffEffect = "Gull-Wing, +100% init and +5 " + HTMLGenerateSpanFont("Hot", "red") + " resist"; break;
-			case 7:
-				nextBeardBuffEffect = "Space Warlord, +25 Weapon damage and +10% Crit"; break;
-			case 8:
-				nextBeardBuffEffect = "Pointy Wizard, +25 Spell damage and +10% Spell Crit"; break;
-			case 9:
-				nextBeardBuffEffect = "Cowboy, +25 Ranged damage and +50 maximum HP"; break;
-			case 10:
-				nextBeardBuffEffect = "Friendly, +100% meat and +5 " + HTMLGenerateSpanFont("Sleaze", "purple") + " resist"; break;
-			case 0:
-				nextBeardBuffEffect = "Spectacle, +50% item and +5 " + HTMLGenerateSpanFont("Spooky", "grey") + " resist"; break;
+
+	string [int] description;
+	string url = "inventory.php?ftext=daylight+shavings+helmet";
+	description.listAppend("Shave turns off of your sanity!");
+
+	int beardBuff = get_property_int("lastBeardBuff");
+	int shavingBuffCounter;
+	if (beardBuff == 0)
+	{
+		shavingBuffCounter = 0;
+	}
+	else {
+		shavingBuffCounter = beardBuff - 2665;
+	}
+
+	// Shaving buff prediction
+	string [string] nextBeardBuff = getNthClassBuff(shavingBuffCounter + 1);
+
+	string nextBeardBuffDescription;
+	string nextBeardBuffEffect;
+	foreach nextBeardBuffName in nextBeardBuff {
```

Comment:
i know this isn't enormously complicated but big props on making this code much more concise, snappy, and straightforward. i appreciate the lil generators and stuff you've got here, it works quite well.